### PR TITLE
New Feature: Check For Update

### PR DESCRIPTION
**Pull request by paulpv**
This code requires an update server to be running at https://distroav.org/api/update.
The update server is a Firebase Function with code at https://github.com/DistroAV/firebase/blob/main/functions/index.js

Much of this PR duplicates PR #960.
I would like to merge this PR in as a 4.x version and then PR #960 as a 6.x that support NDI 6.x
I will resolve any conflicts when I merge master into #960

### CODE REVIEW

File: cmake/windows/resources/installer-Windows.iss.in
Comment (Trouffman) [inline cmake/windows/resources/installer-Windows.iss.in:1]: I investigated this to see if the correct way is AppId={@UUID_APP@} or AppId={{@UUID_APP@}}.
The double bracket option is invalid.
Good catch!

File: src/forms/output-settings.cpp
Comment (Trouffman) [inline src/forms/output-settings.cpp:1]: Is this in any relation with the "Check for update" feature?

File: src/obs-ndi-filter.cpp
Comment (Trouffman) [inline src/obs-ndi-filter.cpp:1]: Is this in any relation with the "Check for update" feature?

File: cmake/windows/resources/installer-Windows.iss.in
Comment (paulpv) [reply cmake/windows/resources/installer-Windows.iss.in:1]: Thanks for bringing that up!
I did discuss this w/ the OBS team:
https://discord.com/channels/348973006581923840/374636084883095554/1240927807828267090

> PatTheMav — 05/17/2024 2:27 PM
> no need for DMs, the answer is "I have no clue"
> I just replaced the value with a CMake replacement, no idea what is "correct" there
> PatTheMav — 05/17/2024 2:29 PM
> hmmmm so the Arc team has that covered it seems.. https://speakinginswift.substack.com/p/swift-meet-winrt
> Swift, meet WinRT
> Open sourcing our Swift bindings generator for WinRT - and an end-to-end sample application for anyone looking > to build a modern Windows application in Swift.
> [Swift, meet WinRT](https://speakinginswift.substack.com/p/swift-meet-winrt)
> RytoEX — 05/17/2024 2:31 PM
> Anyway, this is (sort of) covered in the Inno docs:
> A "{" character is treated as the start of the constant. If you want to use that actual character in a place where > constants are supported, you must use two consecutive "{" characters. (You do not need to double "}" characters.)
>  
> https://jrsoftware.org/ishelp/
> So I'd guess that the desired constant value is:
> {CD703FE5-1F2C-4837-BD3D-DD840D83C3E3}
> 
> or:
> {@UUID_APP@}
> 
> and doubling the { character achieves that.
> Or it's a mistake, but I'm in no condition at the moment to dig further.
> RytoEX — 05/17/2024 2:38 PM
> I guess the real question is: are you having any issues stemming from this syntax?
> If not, then it's probably working as intended.
> pvhaus (PST/UTC-8) — 05/17/2024 8:20 PM
> No, it just caught my eye as a little odd. I am trying some builds on my plug-in (obs-ndi) to see if it makes any difference (fixes or breaks anything)

So, that made me second guess that change, but now you are making be second guess my second guess.

File: src/obs-ndi-filter.cpp
Comment (paulpv) [reply src/obs-ndi-filter.cpp:1]: Yes. It is a little bit of a stretch, but for the update testing I chose to `rehostUrl` all URLs, and that required me to touch this code.

File: cmake/windows/resources/installer-Windows.iss.in
Comment (Trouffman) [reply cmake/windows/resources/installer-Windows.iss.in:1]: The double bracket (on each side) seems invalid :D
Now I am double guessing that the first double bracket might be okay!

File: data/locale/fr-FR.ini [hunk 1]
```diff
@@ -30,8 +30,48 @@ NDIPlugin.Menu.OutputSettings="NDI™ - Sortie principale"
...
+NDIPlugin.Update.RemindMeLater="Me le rappeler plus tard"
+NDIPlugin.Update.InstallUpdate="Installer la mise à jour"
+NDIPlugin.Update.CheckingForUpdate.Text="Vérification de la mise à jour de %1..."
+NDIPlugin.Update.CheckingForUpdate.Cancel="Annuler"
+NDIPlugin.Update.CheckingForUpdate.Error.Title="Erreur de vérification de mise à jour"
+NDIPlugin.Update.CheckingForUpdate.Error.Text="Erreur lors de la vérification de mise à jour :<br><pre>%1</pre>"
+NDIPlugin.Update.NoUpdateAvailable="Aucune mise à jour de %1 disponible"
+NDIPlugin.Update.YouAreUpToDate="Vous avez déjà la dernière version %1 de %2."
+
+NDIPlugin.Donate="Veuillez envisager de faire un don pour soutenir ce projet :"
```
Comment (Trouffman) [inline data/locale/fr-FR.ini:58]: Suggested : 
NDIPlugin.Donate="Faites un don pour soutenir ce projet :"
More direct & to the point.

Comment (paulpv) [reply data/locale/fr-FR.ini:58]: No "Please"? :)
I did just notice that stray whitespace. :/

**Pull request merged**